### PR TITLE
Handlebars template fixes

### PR DESCRIPTION
**Pull request by JanJakes**
When setting up a custom theme, I discovered multiple issues in the Handlebars templates. I'm adding fixes for all of them. I also added a few optional improvements (mainly related to using Prettier with `angular` parser to format `.hbs` files), happy to update/drop any of them, if needed. The commits describe the changes:

1. Fix wrong Subscribe/Account button condition 
2. Fix unmatched tag in signup.hbs 
3. Replace invalid `{{^if ...}}` with `{{#unless ...}}`
4. Remove unmatched closing ")" in sizes attribute
5. Use single quotes in helper arguments 
6. Move conditions inside attribute values 
7. Remove tags from comments (they can confuse some parsers) 
8. Remove trailing whitespace
9. Do not render code comments as HTML comments

### CODE REVIEW

**Comment by JohnONolan** [comment]
I realise this is an old PR - so you probably figured this out already - but `{{^if}}` is not invalid syntax - this is correct 
https://docs.ghost.org/themes/helpers/functional/if#if

**Comment by JanJakes** [comment]
@JohnONolan Getting a response after almost two years, I don't remember all the details, but I do wonder why you're commenting just on point 3 out of all the issues I was trying to help to fix, and closing the issue without any explanation.

Thanks for the clarification on the `{{^if}}` syntax. I can see that it's valid in Ghost, but it is still not a valid [Handlebars.js](https://handlebarsjs.com/) syntax. That can cause problems in editor and IDE support, and it breaks some formatters and similar tools.

Now, what about the other issues? Were they already fixed, and are they not relevant anymore? To summarize:
1. This is the `{{^if}}` syntax. I was only making it Handlebars.js compatible.
2. **[Fix unmatched tag in signup.hbs](https://github.com/TryGhost/Starter/pull/90/commits/817ea7043dc868ab2a89ce31ec7f6e57caa78d2e) — this is invalid HTML.**
3. This is the `{{^if}}` syntax. I was only making it Handlebars.js compatible.
4. **[Remove unmatched closing ")" in sizes attribute](https://github.com/TryGhost/Starter/pull/90/commits/ac31a956324f4ebaa912628a7950f5d74209aa00) — this is invalid HTML.**
5. Code style. I was only making it Angular HBS parser compatible.
6. Code style. I was only making it Angular HBS parser compatible.
7. Code style. I was only making it Angular HBS parser compatible.
8. Improve code style — remove accidental whitespace.

In summary, 2 and 4 fix invalid HTML being produced by the template code, and all the other small changes make the templates fully Handlebars-compatible, which is great for tooling and editor support. I guess my questions are:
- Are the invalid HTML fixes no longer relevant? Have they been already resolved in the meantime?
- Are you not interested in compatibility with standard Handlebars tooling?

Actually, [I just checked](https://github.com/TryGhost/Starter/commits/main/), and there are almost no commits since I created this PR. It appears that the starter theme still produces invalid HTML. Are you not interested in fixing at least that part?
